aikau/src/main/resources/alfresco/layout/AlfTabContainer.js

```diff
@@ -589,6 +593,26 @@ define(["dojo/_base/declare",
          {
             array.forEach(payload.widgets, lang.hitch(this, this.addWidget));
          }
+
+         // See AKU-583...
+         // It is possible to include a "publishOnAdd" attribute when creating tabs that define an additional publication to perform
+         // after the tab is added...
+         if (payload.publishOnAdd)
+         {
+            var publication = payload.publishOnAdd;
+            if (publication.publishTopic)
+            {
+               this.alfPublish(publication.publishTopic,
+                               publication.publishPayload,
+                               publication.publishGlobal,
+                               publication.publishToParent,
+                               publication.publishScope);
+            }
+            else
+            {
+               this.alfLog("warn", "A request to add a tab included a following publication, but it did not include a 'publishTopic' attribute", payload, this);
+            }
+         }
```

Comment:
Should we be defensively coding to deal with the possibility that this "on add" publish needs all of the widgets to have successfully loaded and been started beforehand?